src/conditions/base/index.ts

```diff
@@ -1,4 +1,40 @@
-export { Condition } from './condition';
-export { ContractCondition } from './contract';
-export { RpcCondition } from './rpc';
-export { TimeCondition } from './time';
+import {
+  CompoundConditionProps,
+  compoundConditionSchema,
+} from '../compound-condition';
+import { Condition } from '../condition';
+
+import { ContractConditionProps, contractConditionSchema } from './contract';
+import { RpcConditionProps, rpcConditionSchema } from './rpc';
+import { TimeConditionProps, timeConditionSchema } from './time';
+
+// Exporting classes here instead of their respective schema files to
+// avoid circular dependency on Condition class.
+
+export class CompoundCondition extends Condition {
+  constructor(value: CompoundConditionProps) {
+    super(compoundConditionSchema, value);
+  }
+}
+
+export class ContractCondition extends Condition {
+  constructor(value: ContractConditionProps) {
+    super(contractConditionSchema, value);
+  }
+}
+
+export class RpcCondition extends Condition {
+  constructor(value: RpcConditionProps) {
+    super(rpcConditionSchema, value);
+  }
+}
+
+export class TimeCondition extends Condition {
+  constructor(value: TimeConditionProps) {
+    super(timeConditionSchema, value);
+  }
+}
+
+export { type ContractConditionProps } from './contract';
+export { type RpcConditionProps } from './rpc';
+export { type TimeConditionProps } from './time';
```

Comment:
Should we also export CompoundConditionProps?

Reply:
It's already exported from here: https://github.com/nucypher/nucypher-ts/pull/267/files#diff-c69f0d621a87085de1005a17d7157e7bf6fe13316cf6003268fff6af266fee11R11